hadoop-hdds/server-scm/src/test/java/org/apache/hadoop/ozone/container/common/TestEndPoint.java

```diff
@@ -473,10 +473,10 @@ public void testRegisterNoContainerID() throws Exception {
   @Test
   public void testRegisterRpcTimeout() throws Exception {
     final long rpcTimeout = 1000;
-    final long tolerance = 200;
-    scmServerImpl.setRpcResponseDelay(1500);
+    final long tolerance = 900;
+    scmServerImpl.setRpcResponseDelay(2500);
     long start = Time.monotonicNow();
-    registerTaskHelper(serverAddress, 1000, false).close();
+    registerTaskHelper(serverAddress, (int) rpcTimeout, false).close();
     long end = Time.monotonicNow();
```

Comment:
I think the test should check elapsed time only for `endpointTask.call()`:

```java
    final int rpcTimeout = 1000;
    final int tolerance = 200;
    scmServerImpl.setRpcResponseDelay(rpcTimeout + tolerance * 2);
    OzoneConfiguration conf = SCMTestUtils.getConf(tempDir);
    try (EndpointStateMachine rpcEndPoint = createEndpoint(conf, serverAddress, rpcTimeout)) {
      rpcEndPoint.setState(EndpointStateMachine.EndPointStates.REGISTER);
      RegisterEndpointTask endpointTask = getRegisterEndpointTask(false, rpcEndPoint, conf);
      long start = Time.monotonicNow();
      endpointTask.call();
      long end = Time.monotonicNow();
      assertThat(end - start)
          .isGreaterThanOrEqualTo(rpcTimeout)
          .isLessThanOrEqualTo(rpcTimeout + tolerance);
    } finally {
      scmServerImpl.setRpcResponseDelay(0);
    }
```

We'd need some refactoring (extract methods) for that.

Reply:
> the test should check elapsed time only for endpointTask.call()

I agree, we might only need to capture the latency of the `endpointTask.call()`.

Reply:
Thanks for the reviews! I have updated the PR to refactor the test. Now the latency of only endpointTask.call() is measured. Could you please review the updated patch? @adoroszlai @tanvipenumudy